kafka-connect/src/main/java/io/tabular/iceberg/connect/channel/Coordinator.java

```diff
@@ -255,6 +305,127 @@ private void commitToTable(
     }
   }
 
+  private static class PartitionOffset {
+
+    private final int partition;
+    private final long offset;
+
+    PartitionOffset(int partition, long offset) {
+
+      this.partition = partition;
+      this.offset = offset;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+      if (this == o) {
+        return true;
+      }
+      if (o == null || getClass() != o.getClass()) {
+        return false;
+      }
+      PartitionOffset that = (PartitionOffset) o;
+      return partition == that.partition && offset == that.offset;
+    }
+
+    @Override
+    public int hashCode() {
+      return Objects.hash(partition, offset);
+    }
+
+    @Override
+    public String toString() {
+      return "PartitionOffset{" + "partition=" + partition + ", offset=" + offset + '}';
+    }
+  }
+
+  private List<Envelope> deduplicateEnvelopes(
```

Comment:
**minor** Would pull out to a class/static method for ease of testing, coordinator is already a complicated class

Reply:
Good idea, done.